stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -332,6 +333,31 @@ static int _addImageProtocolConformances(struct dl_phdr_info *info,
   dlclose(handle);
   return 0;
 }
+#elif defined(__CYGWIN__)
+static int _addImageProtocolConformances(struct dl_phdr_info *info,
```

Comment:
@modocache , @gribozavr , How about this snippet code [unify.cpp](https://github.com/tinysun212/swift-cygwin/blob/88abb1b42278a66593e85f8796c15b5520b81ba8/misc/unify.cpp) for  dylib callbacks?